core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -1,14 +1,27 @@
 package com.mesosphere.usi.core
 
-import com.mesosphere.usi.core.models.PodSpec
+import akka.NotUsed
+import akka.stream.BidiShape
+import akka.stream.scaladsl.{BidiFlow, Broadcast, GraphDSL}
+import com.mesosphere.usi.models.{Mesos, SpecMessage, StatusMessage}
 
-import scala.concurrent.Future
+object Scheduler {
+  def graph: BidiFlow[SpecMessage, StatusMessage, Mesos.Event, Mesos.Call, NotUsed] = {
```

Comment:
Could you add docs to this method? It seems to be of importance but I have no idea what's really happening here?

Reply:
it helped me to draw it down

```    * {{{
    *                 +------+
    *  SpecMessages ~>|      |~> StatusMessages
    *                 | bidi |
    *  Mesos.Call <~  |      |<~ Mesos.Event
    *                 +------+
    * }}}```

but I still don't totally get the flow of data here. I would expect SpecMessage triggers Mesos.Call and Mesos.Event triggers StatusMessage but this does not seem to be the case right now. @timcharper

Reply:
also I guess this is not the most highest level interface, right? The one framework will interact... Because then I would expect it to be just `SpecMessages ~> StatusMessages` I think I am still puzzled by the fact that we expose things that are to me on totally different levels in one component - that's the outside USI interface as well as stuff happening internally (mesos calls and events). Could you maybe clarify that?

Reply:
Could we add the flow chart to the docs above. If it uses PlantUML or Mermaid syntax I'm more than happy to render them in our scaladocs.